presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -0,0 +1,323 @@
+import static com.facebook.presto.spi.security.AccessDeniedException.denySelectTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denySelectView;
+import static com.facebook.presto.spi.security.AccessDeniedException.denySetCatalogSessionProperty;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyShowSchemas;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyShowTables;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.util.Objects.requireNonNull;
+
+public class FileBasedSystemAccessControl
+        implements SystemAccessControl
+{
+    public static final String NAME = "file";
+
+    private final List<CatalogAccessControlRule> catalogRules;
+
+    private FileBasedSystemAccessControl(List<CatalogAccessControlRule> catalogRules)
+    {
+        this.catalogRules = catalogRules;
+    }
+
+    public static class Factory
+            implements SystemAccessControlFactory
+    {
+        private static final String CONFIG_FILE_NAME = "security.config-file";
+
+        @Override
+        public String getName()
+        {
+            return NAME;
+        }
+
+        @Override
+        public SystemAccessControl create(Map<String, String> config)
+        {
+            requireNonNull(config, "config is null");
+
+            String configFileName = config.get(CONFIG_FILE_NAME);
+            checkState(
+                    configFileName != null,
+                    "Configuration option %s not found while loading file based system access control", CONFIG_FILE_NAME);
+
+            try {
+                Path path = Paths.get(configFileName);
+                if (!path.isAbsolute()) {
+                    path = path.toAbsolutePath();
+                }
+                return new FileBasedSystemAccessControl(
+                        jsonCodec(FileBasedSystemAccessControlRules.class).fromJson(Files.readAllBytes(path)).getCatalogRules());
+            }
+            catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    @Override
+    public void checkCanSetUser(Principal principal, String userName)
+    {
+    }
+
+    @Override
+    public void checkCanSetSystemSessionProperty(Identity identity, String propertyName)
+    {
+    }
+
+    @Override
+    public Set<String> filterCatalogs(Identity identity, Set<String> catalogs)
+    {
+        ImmutableSet.Builder<String> filteredCatalogs = ImmutableSet.builder();
+        for (String catalog : catalogs) {
+            if (canAccessCatalog(identity, catalog)) {
+                filteredCatalogs.add(catalog);
+            }
+        }
+        return filteredCatalogs.build();
+    }
+
+    @Override
+    public void checkCanCreateSchema(Identity identity, CatalogSchemaName schema)
+    {
+        if (!canAccessCatalog(identity, schema.getCatalogName())) {
```

Comment:
we talked offline and these canAccessCatalog checks are necessary (for now) -- if you keep them and can't fix filterCatalogs(), throw a denyAccessCatalog() exception instead of denyCreateSchema() to make it clear what the issue is.